src/QueryStrategies/BatchQueryStrategies/EnumHierarchicalBatchQs.jl

```diff
@@ -0,0 +1,49 @@
+struct EnumHierarchicalBatchQs <: MultiObjectiveBatchQs
+    model::SVDD.OCClassifier
+    inf_measure::SequentialPQs
+    rep_measure::Function
+    div_measure::Function
+    k::Int
+
+    function EnumHierarchicalBatchQs(model::SVDD.OCClassifier, informativeness::SequentialPQs; representativeness::Symbol=nothing, diversity::Symbol=nothing,
+        k::Int=0)::EnumHierarchicalBatchQs
+        # check basic params
+        (model == nothing) && throw(ArgumentError("No model specified."))
+        (k < 1) && throw(ArgumentError("Invalid batch size k=$(k)."))
+
+        representativeness_measure = get_rep_measure(representativeness)
+        diversity_measure = get_enumerative_div_measure(diversity)
+        return new(model, informativeness, representativeness_measure, diversity_measure, k)
+    end
+end
+
+"""
+Idea: Find 4*k elements with largest representativeness,
+    select 2*k elements with largest informativeness,
+    select batch of size k with maximum diversity
+"""
+function select_batch(qs::EnumHierarchicalBatchQs, x::Array{T, 2}, labels::Dict{Symbol, Vector{Int}}, candidate_indices::Vector{Int})::Vector{Int} where T <: Real
+    num_observations = length(candidate_indices)
+    if num_observations <= qs.k
+        return candidate_indices
+    end
+
+    # representativeness
+    rep_scores = qs.rep_measure(qs.model, x, labels, candidate_indices)
+    most_representative_indices = candidate_indices[sortperm(rep_scores, rev=true)[1:min(4*qs.k, num_observations)]]
+
+    #informativeness
+    inf_scores = qs_score(qs.inf_measure, x, labels)[most_representative_indices]
```

Comment:
This would be more efficient
```julia
inf_scores = qs_score(qs.inf_measure, x[:,most_representative_indices], labels)
```

Reply:
We cannot take this shortcut when using a sequential QS where the informativeness of a single observation depends on other observations. Examples are hybrid QS such as `BoundaryNeighborCombinationPQs`.